core/src/main/java/org/kiwix/kiwixmobile/core/reader/ZimFileReader.kt

```diff
@@ -127,13 +126,10 @@ class ZimFileReader constructor(
     return loadContent(uri)
   }
 
-  fun readMimeType(uri: String) = uri.filePath.let {
-    it.mimeType?.takeIf(String::isNotEmpty) ?: mimeTypeFromReader(it) ?: DEFAULT_MIME_TYPE
-  }.also { Log.d(TAG, "getting mimetype for $uri = $it") }
-
-  private fun mimeTypeFromReader(it: String) =
-    // Truncate mime-type (everything after the first space and semi-colon(if exists)
-    jniKiwixReader.getMimeType(it.filePath)?.replace("^([^ ]+).*$", "$1")?.substringBefore(";")
+  fun readMimeType(uri: String): String = getContentAndMimeType(uri)
+    .second.replace("^([^ ]+).*$", "$1").substringBefore(";").also {
+      Log.d(TAG, "getting mimetype for $uri = $it")
+    }
```

Comment:
@MohitMaliDeveloper It would be better if we add one more test case that replaces the string.